### ISSUE TITLE
Current Unit* model tells more than we should know in the game

### ISSUE THREAD

**Issue by heinermann** [bwapi/bwapi#191]
_From [hero...@gmail.com](https://code.google.com/u/102610364070870861899/) on January 29, 2010 14:15:25_

I understand that currently a unit will have a unique address Unit*.

So here is the possible situation:

When my bot send a SCV to scout a certain area, it sees 4 dragoons, which 
is 4 Unit*. of course, the SCV get killed soon. Later, another SCV goes to 
the main base of the enmemy, and sees 6 dragoons.
What does a person know in this case? a person, in this case, only knows 
that the enemy has at least 6 dragoons.
However, my bot can compare the address of these 6 dragoons with those 4 
dragoons, that effectively let my bot found out "OMG, they got 10 dragoons, 
I should not push out."

In summary, current model enables the bot to do some cheating work even the 
bot doesn't hack at all.

_Original issue: http://code.google.com/p/bwapi/issues/detail?id=192_
